spot_driver/CMakeLists.txt

```diff
@@ -71,35 +93,98 @@ target_include_directories(spot_image_publisher_component
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:include>
 )
-target_link_libraries(spot_image_publisher_component PUBLIC spot_api)
+target_link_libraries(spot_image_publisher_component PUBLIC spot_image_publisher)
 
 rclcpp_components_register_node(
   spot_image_publisher_component
   PLUGIN "spot_ros2::images::SpotImagePublisherNode"
   EXECUTABLE spot_image_publisher_node_component)
 
+###
+# Spot state publisher
+###
+
+# Base library container SpotRobotStatePublisher
+add_library(spot_robot_state_publisher
+   src/robot_state/spot_robot_state_middleware_handle.cpp
+   src/robot_state/spot_robot_state_publisher.cpp
+   src/robot_state/spot_robot_state_publisher_node.cpp
+)
+
+target_include_directories(spot_robot_state_publisher
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
+
+target_link_libraries(spot_robot_state_publisher PUBLIC spot_api)
+set_property(TARGET spot_robot_state_publisher PROPERTY POSITION_INDEPENDENT_CODE ON)
+ament_target_dependencies(spot_robot_state_publisher PUBLIC ${THIS_PACKAGE_INCLUDE_ROS_DEPENDS})
+
+# Create executable to allow running SpotRobotStatePublisherNode directly as a ROS 2 node
+add_executable(spot_robot_state_publisher_node src/robot_state/spot_robot_state_publisher_node_main.cpp)
+target_link_libraries(spot_robot_state_publisher_node PUBLIC spot_robot_state_publisher)
+target_include_directories(spot_robot_state_publisher_node
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
+
+# Register a composable node to allow loading SpotRobotStatePublisherNode in a component container
+add_library(spot_robot_state_publisher_component SHARED src/robot_state/spot_robot_state_publisher_component.cpp)
+target_include_directories(spot_robot_state_publisher_component
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
+target_link_libraries(spot_robot_state_publisher_component PUBLIC spot_robot_state_publisher)
+ament_target_dependencies(spot_robot_state_publisher_component PUBLIC rclcpp_components)
+
+rclcpp_components_register_node(
+  spot_robot_state_publisher_component
+  PLUGIN "spot_ros2::SpotRobotStatePublisherNode"
+  EXECUTABLE spot_robot_state_publisher_node_component)
+
+###
+# Spot IK
+###
+
+# Base library containing the SpotInverseKinematics class
+add_library(spot_inverse_kinematics
```

Comment:
Why splitting the spot API library into multiple libraries? Is it worth value in terms of later maintenance?

Reply:
I combined these libraries into a single target. I was thinking that it makes the build a little more granular since changing one of these libraries doesn't require rebuilding the other libraries plus the things that depend on those libraries, but in practice it doesn't make a big difference.